context.go

```diff
@@ -31,6 +31,13 @@ func (nc *Conn) RequestWithContext(ctx context.Context, subj string, data []byte
 	if nc == nil {
 		return nil, ErrInvalidConnection
 	}
+	// Check whether the context is done already before making
+	// the request.
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+	}
```

Comment:
Without this check, if a message was received fast enough then it would have succeeded sometimes even if the `context.Done()` channel was already closed because of `select` randomly choosing to receive the channel:
https://github.com/nats-io/go-nats/blob/2f0e8883ae354dfdeb89680e43c8c6918d34b815/context.go#L66-L84